File: solution/day04/Routine.Tests/Fakes/EmailService.cs
using FluentAssertions;

namespace Routine.Tests.Fakes
{
    public class EmailService : IEmailService
    {
        private bool _emailRead;
        public void ReadNewEmails() => _emailRead = true;

        public void HasBeenRead() => _emailRead.Should().BeTrue();
    }
}

File: exercise/ts/day04/tests/doubles/emailService.ts
import { EmailService } from '../../src/routine/emailService';

export class EmailServiceDouble implements EmailService {
    private _readNewEmailsCalled: boolean = false;

    get readNewEmailsCalled(): boolean {
        return this._readNewEmailsCalled;
    }
    readNewEmails() {
        this._readNewEmailsCalled = true;
    }
}
